hypervisor/src/hypervisor/linux.rs

```diff
@@ -119,13 +112,26 @@ impl Hypervisor {
 
         let mut frame_start = Instant::now();
 
+        // retain the first frame start as our sytems t0
+        if self.t0.is_none() {
+            self.t0 = Some(frame_start);
+        }
+
         let sys_time = SYSTEM_START_TIME
             .get()
             .ok_or_else(|| anyhow!("SystemTime was not set"))
             .lev_typ(SystemError::Panic, ErrorLevel::ModuleInit)?;
         sys_time.write(&frame_start).lev(ErrorLevel::ModuleInit)?;
         sys_time.seal_read_only().lev(ErrorLevel::ModuleInit)?;
         loop {
+            // if we are not ment to execute any longer, terminate here
+            match self.terminate_after {
+                Some(terminate_after) if frame_start - self.t0.unwrap() > terminate_after => {
+                    quit::with_code(0)
```

Comment:
Does this forceful termination cleans up the cgroups? Or does it leak? For me, the latter one looks more likely to occur.

Reply:
@Steav005 Your expertise is in question here. AFAIK (at least telling from the log outputs) the Destructors _are_ ran. From [`docs.rs/quit`](https://docs.rs/quit/latest/quit/):

> This crate allows cleanly exiting a program using a custom exit code, without the drawbacks of [process::exit](https://doc.rust-lang.org/std/process/fn.exit.html). Destructors will be called as usual, and the stack will be unwound to the main function.

Reply:
Normally, running destructors is not enough to clean up cgroup, BUT we implemented [`Drop`](https://doc.rust-lang.org/std/ops/trait.Drop.html) for the hypervisor which handles the clean-up should the hypervisor struct be dropped.
https://github.com/aeronautical-informatics/apex-linux/blob/137e622ab2baeb41c187cab5b3ccbd2b67342523/hypervisor/src/hypervisor/linux.rs#L144
So calling `quit::with_code(0)` is enough, because it [unwinds the stack](https://doc.rust-lang.org/nomicon/unwinding.html) for us.  
But i am with @emilengler in that logging should be added.  
We should also think about exit codes to use for the hypervisor in different scenarios.
 - Ctrl+C 
 - Timeout
 - Health Monitor exit